src/Persistence/MassTransit.Azure.Cosmos/Configuration/Configuration/CosmosSagaRepositoryConfigurator.cs

```diff
@@ -62,6 +69,8 @@ public string CollectionId
             }
         }
 
+        public string CosmosClientName { get; set; }
```

Comment:
I think a method, as I mentioned in the discussion, would be preferred here, similar to all the others. For example:

```cs
public void UseClientFactory(string name) {}
```

That would signal the use of actually calling GetRequiredService, to ensure the explicit request to use the client factory is actually sourced by the container. It's sort of transparent and would fall back to the default behavior otherwise, which might not be intended.

Reply:
Also, a doc header indicating the reason, with a cref to the interface to use.

Reply:
Added the method as suggested and started using `GetRequiredService` if the client name was set.

There is a basic implementation of an `ICosmosClientFactory` to create a static client based on the endpoint and key. Right now it's part of the main library, but only used in the tests. I debated if this should exist in the library or just the test code, but ended up with the library out of utility (even though it uses the same creation as the existing one without the DI container). Do you think this basic implementation should be added to the DI container as part of the implementation using `TryAddSingleton` (or the appropriate container equivalent)? I'm leaning towards no since the use of `UseClientFactory` is very explicit and the caller is expected to make sure a client factory has actually been registered, but I'd like your input on this one.